packages/analytics/amplify_analytics_pinpoint_dart/lib/src/exception/pinpoint_analytics_exception.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+const _fileIssueMessage =
+    'This exception is not expected. Please try again. If the exception persists, please file an issue at https://github.com/aws-amplify/amplify-flutter/issues';
+
+/// {@template amplify_analytics_pinpoint_dart.pinpoint_exception}
+/// Represents exceptions that may be thrown calling Pinpoint Analytics plugin APIs.
+/// {@endtemplate}
+class PinpointException extends AnalyticsException {
```

Comment:
That makes sense.  

Looking at the code, it seems Storage is the only category that's throwing plugin specific exceptions (S3Exceptions) now.  The logic of your comment might apply there too.  

Though for storage, the user might want to catch the individual exception types while in the case for Analytics here there really isn't much unique logic that I'd anticipate customers would write for a BadRequest/NotFound/Forbidden case beyond seeing it in their debugging. 

I'll update Analytics. 


However, perhaps Storage should also change and now have Storage vs. S3 exceptions as well. 

In fact, shouldn't we throw the specific error types for Storage? 

For example: 

S3IncompleteDownloadException instead of a S3Exception with an exception message of incomplete download string? 

I'd anticipate customers might actually want to have custom logic based on that specific case. 

cc @HuiSF

Reply:
Good point from you both.
The `S3Exception` serves as a collection of the exception parser function but ultimately they are `StorageException`. Throwing specific exception variants here is unnecessary, only the specific storage exception types defined in core should be thrown. I can find some time to update the `S3Exception`.

Reply:
Sounds good.  If you are busy I can also find time to redo the S3Exception and make a PR for your review @HuiSF.  I'll see how my other work goes and then followup with you in DM.